rules/source-control-tools/rule.md

```diff
@@ -0,0 +1,143 @@
+created: 2024-01-30T13:44:07.470Z
+guid: 87b1677f-72c8-4b6f-9976-d3cfab075272
+---
+Version control is an essential tool for software developers. It enables tracking changes made to a project's source code, facilitates collaboration among developers, and ensures the project's consistency.
+
+<!--endintro-->
+
+Version control is an essential tool for software developers. It enables tracking changes made to a project's source code, facilitates collaboration among developers, and ensures the project's consistency.
+
+### Advantages of Version Control Software
+
+1. **Project Consistency Assurance:** it keeps a record of all changes made to a project's source code. This allows reverting to a previous version of the code in case of problems.
+2. **Facilitating Collaboration:** it allows developers to work simultaneously on the same project without the risk of conflict. They provide a centralized repository for storing and managing source code.
+3. **Improving Productivity:** it can enhance developers' productivity by automating tasks like creating branches and tracking changes, and by abstracting the use of the console for most common actions.
+4. **Reducing Risks:** it can help reduce risks associated with software development. They can assist in identifying and fixing bugs more quickly, and they can also help protect projects from data loss.
+5. **Improving Code Quality:** it can help improve code quality by encouraging developers to write cleaner and more maintainable code.
+6. **Promoting Communication:** it can help promote communication among developers. They provide a way for developers to share their changes and collaborate on projects.
+
+### Disadvantages of Version Control Software
+
+1. **Complexity:** Version control software can be complex to learn and use. This can be a barrier for beginner developers.
+2. **Cost:** Commercial version control software can be expensive. However, many free open-source solutions are available.
+3. **Performance:** Version control software can negatively impact the performance of the systems they are used on. This is particularly true for large projects.
+4. **Security:** Version control software can be vulnerable to attacks. It is important to take security measures to protect the data stored in a version control repository.
+
+::: greybox
+**🧐 Did You Know?**  
+One of the earliest version control systems, named **SCCS (Source Code Control System)**, was developed by Marc J. Rochkind at Bell Labs in 1972. At that time, the need to efficiently manage changes to source code was just beginning to emerge in the software industry.  
+This system enabled the tracking of revisions for individual files, a significant innovation for its time. It's from these early initiatives that modern version control tools like Git and Mercurial have evolved, offering far more advanced features and collaborative code management capabilities.
+:::
+
+### [GitKraken](https://www.gitkraken.com/)
```

Comment:
Hey @PothieuG 

The headings already have an auto anchor link to them... so we try to avoid they being a link

- [ ] 1. Can you move the link from the heading here to the 1st word in the next paragraph
- [ ] 2. Do the same for other software (move link to the 1st mention of the product name in the paragraphs)